automation_tools/__init__.py

```diff
@@ -2027,27 +1979,71 @@ def product_install(distribution, certificate_url=None, selinux_mode=None, sat_v
 
     :param str distribution: product distribution wanted to install
     :param str certificate_url: where to fetch a fake certificate.
-    :param sat_version: Indicates which satellite version should be installed
+    :param str sat_version: Indicates which satellite version should be installed
 
     """
+    distribution = distribution.lower()
+    # Strip useless 'satellite6-' prefix if there is such
+    distribution = distribution[11:] if distribution.startswith('satellite6-') else distribution
     # Fetch the Satellite Version information.
     sat_version = sat_version or os.environ.get('SATELLITE_VERSION')
+    selinux_mode = selinux_mode or os.environ.get('SELINUX_MODE', 'enforcing')
+    admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
 
     # Command-line arguments are passed in as strings.
     if isinstance(test_in_stage, str):
         test_in_stage = (test_in_stage.lower() == 'true')
 
     install_tasks = {
-        'satellite6-beta': cdn_install,
-        'satellite6-cdn': cdn_install,
-        'satellite6-downstream': downstream_install,
-        'satellite6-repofile': repofile_install,
-        'satellite6-activationkey': ak_install,
-        'satellite6-iso': iso_install,
-        'satellite6-upstream': upstream_install,
-        'satellite6-koji': upstream_install,
+        'beta': cdn_install,
+        'cdn': cdn_install,
+        'downstream': downstream_install,
+        'repofile': repofile_install,
+        'activationkey': ak_install,
+        'iso': iso_install,
+        'upstream': upstream_install,
+        'koji': upstream_install,
     }
-    distribution = distribution.lower()
+
+    # install task params population
+    task_params = {
+        'run_katello_installer': False,
+        'admin_password': admin_password,
+    }
+    if distribution in ('upstream', 'koji'):
+        task_params.update({'distribution': distribution})
+    else:
+        task_params.update({'sat_version': sat_version})
+
+    if distribution == 'repofile':
+        url = os.environ.get('REPO_FILE_URL')
```

Comment:
Why this is not converted ?

Reply:
this PR deals with using env vars in subtasks, for product_install I would raise another PR. Ok?

I just dont like reading env vars on and on esp. in sub tasks when they should be passed as params